OsmAnd/src/net/osmand/plus/voice/CommandBuilder.java

```diff
@@ -30,6 +30,7 @@
 	protected static final String C_REACHED_POI = "reached_poi";
 	protected static final String C_THEN = "then";
 	protected static final String C_SPEAD_ALARM = "speed_alarm";
+	protected static final String C_SPEED_CAMERA_ALARM = "speed_camera_alarm";
```

Comment:
This approach will require to change all voice configuration that could be too much. So incremental approach or check could be implemented probably.

Reply:
I have added corresponding fallback in the voice configuration files as part of the related PR https://github.com/osmandapp/OsmAnd-resources/pull/724/files

For all languages that contains
`dictionary["distance"] + (tts ? " - " : " ") + distance(dist) ` as part of the `function route_recalc(dist, seconds)` i have added implementation, once this part of announcement will be same. As well as the part of announcement with the speed limit will be the same with corresponding announcement from "regular" speed limit alarm. For all other languages I have added fallback to original announcement without speed / distance.

Reply:
Only one thing that I am not sure, it is how to force voice configuration files update before (or during) latest version installation. It seems that the voice files downloaded separately from the application itself. Probably you can suggest how to check if the latest voice files are available on device (or if voice file contains corresponding function) from the CommandBuilder / JsCommandBuilder, I have not found something similar....

Reply:
Voice files are updated automatically checked each android version update (probably you have same android version number).

Reply:
We've already implemented before backward compatible voice changes, so old voice packages could continue to work with  ```isJsCommandExists(C_TAKE_EXIT)```

Reply:
@vshcherb thank you for refer the `isJsCommandExists`, have missed it.

So, you recommend to add fallback in Java code even if the voice files will be updated as part of another already prepared PR?

The fallback also will require passing additional argument `String fallbackAttentionType` to the speedCameraAlarm method, to avoid dependency between JsCommandBuilder and AlarmInfoType (the attentionType should be `String.valueOf(AlarmInfoType.SPEED_CAMERA)`), which itself could be slightly "dirty".

Reply:
You can never say if somebody doesn't have custom tts package. Secondly, good practice is to update only packages / languages that you are more or less sure. So in that case to not add commands that don't support it. 

Indeed you need to pass parameter that make sense only for speed camera, so probably making a new method make sense but it should be checked anyway that this method exists and fallback should be called.